### ISSUE TITLE
Executing `NavigationLink2D._edit_get_rect` crashes Godot

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#65431]
### Godot version

4.0.alpha.custom_build. b6d102c7c

### System information

Ubuntu 22.04 - Nvidia GTX 970, Gnome shell 42 X11

### Issue description

When executing
```
	var temp_variable6 = NavigationLink2D.new()
	temp_variable6._edit_get_rect()
```

Godot crashes with this backtrace
```
scene/2d/navigation_link_2d.cpp:116:121: runtime error: member call on null pointer of type 'struct World2D'
scene/2d/navigation_link_2d.cpp:116:121: runtime error: member access within null pointer of type 'struct World2D'

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.alpha.custom_build (4b164b8e4790cfeb597b763123560b59a96458e5)
Dumping the backtrace. Please include this when reporting the bug on: https://github.com/godotengine/godot/issues
[1] godot4s() [0x2647ae8] (/home/runner/work/GodotBuilds/GodotBuilds/godot/platform/linuxbsd/crash_handler_linuxbsd.cpp:56)
[2] /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7f1642cbe520] (??:0)
[3] NavigationLink2D::_edit_get_rect() const (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/2d/navigation_link_2d.cpp:116 (discriminator 1))
[4] void call_with_variant_args_retc_helper<__UnexistingClass, Rect2>(__UnexistingClass*, Rect2 (__UnexistingClass::*)() const, Variant const**, Variant&, Callable::CallError&, IndexSequence<>) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/variant/binder_common.h:731 (discriminator 4))
[5] void call_with_variant_args_retc_dv<__UnexistingClass, Rect2>(__UnexistingClass*, Rect2 (__UnexistingClass::*)() const, Variant const**, int, Variant&, Callable::CallError&, Vector<Variant> const&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/variant/binder_common.h:497)
[6] MethodBindTRC<Rect2>::call(Object*, Variant const**, int, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/object/method_bind.h:568)
[7] Object::callp(StringName const&, Variant const**, int, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/object/object.cpp:733 (discriminator 1))
[8] Object::callv(StringName const&, Array const&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/object/object.cpp:671 (discriminator 1))
[9] void call_with_variant_args_ret_helper<__UnexistingClass, Variant, StringName const&, Array const&, 0ul, 1ul>(__UnexistingClass*, Variant (__UnexistingClass::*)(StringName const&, Array const&), Variant const**, Variant&, Callable::CallError&, IndexSequence<0ul, 1ul>) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/variant/binder_common.h:680 (discriminator 8))
[10] void call_with_variant_args_ret_dv<__UnexistingClass, Variant, StringName const&, Array const&>(__UnexistingClass*, Variant (__UnexistingClass::*)(StringName const&, Array const&), Variant const**, int, Variant&, Callable::CallError&, Vector<Variant> const&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/variant/binder_common.h:464)
[11] MethodBindTR<Variant, StringName const&, Array const&>::call(Object*, Variant const**, int, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/object/method_bind.h:481)
[12] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/gdscript/gdscript_vm.cpp:1644 (discriminator 1))
[13] GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/gdscript/gdscript.cpp:1627)
[14] Object::callp(StringName const&, Variant const**, int, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/object/object.cpp:711 (discriminator 1))
[15] Variant::callp(StringName const&, Variant const**, int, Variant&, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/variant/variant_call.cpp:1048)
[16] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/gdscript/gdscript_vm.cpp:1554)
[17] GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/gdscript/gdscript.cpp:1627)
[18] bool Node::_gdvirtual__process_call<false>(double) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/node.h:237 (discriminator 5))
[19] Node::_notification(int) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/node.cpp:57)
[20] Node::_notificationv(int, bool) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./scene/main/node.h:45 (discriminator 14))
[21] Object::notification(int, bool) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/object/object.cpp:792)
[22] SceneTree::_notify_group_pause(StringName const&, int) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/scene_tree.cpp:917)
[23] SceneTree::process(double) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/scene_tree.cpp:465 (discriminator 4))
[24] Main::iteration() (/home/runner/work/GodotBuilds/GodotBuilds/godot/main/main.cpp:2992)
[25] OS_LinuxBSD::run() (/home/runner/work/GodotBuilds/GodotBuilds/godot/platform/linuxbsd/os_linuxbsd.cpp:538)
[26] godot4s(main+0x4aa) [0x2644b90] (/home/runner/work/GodotBuilds/GodotBuilds/godot/platform/linuxbsd/godot_linuxbsd.cpp:74)
[27] /lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7f1642ca5d90] (??:0)
[28] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7f1642ca5e40] (??:0)
[29] godot4s(_start+0x2e) [0x264462e] (??:?)


```

This example was found by Godot fuzzer - [Qarminer](https://github.com/qarmin/Qarminer), so it is quite unlikelly that this code could be used in real project, but still this should be handled gracefully.

### Steps to reproduce

Above

**Comment by akien-mga** [comment]
CC @DarkKilauea

**Comment by Trashtalk217** [comment]
I was able to reproduce the error in Arch Linux with the following debug information:

```
ERROR: Condition "!is_inside_tree()" is true. Returning: Ref<World2D>()
   at: get_world_2d (scene/main/canvas_item.cpp:773)

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.alpha.custom_build (9142904c248cb4c9de8776f9abe995fc22335c2c)
Dumping the backtrace. Please include this when reporting the bug on: https://github.com/godotengine/godot/issues
[1] /usr/lib/libc.so.6(+0x38a40) [0x7f071b417a40] (??:0)
[2] World2D::get_navigation_map() const (??:0)
[3] NavigationLink2D::_edit_get_rect() const (??:0)
[4] void call_with_variant_args_retc_helper<__UnexistingClass, Rect2>(__UnexistingClass*, Rect2 (__UnexistingClass::*)() const, Variant const**, Variant&, Callable::CallError&, IndexSequence<>) (??:0)
[5] void call_with_variant_args_retc_dv<__UnexistingClass, Rect2>(__UnexistingClass*, Rect2 (__UnexistingClass::*)() const, Variant const**, int, Variant&, Callable::CallError&, Vector<Variant> const&) (??:0)
[6] MethodBindTRC<Rect2>::call(Object*, Variant const**, int, Callable::CallError&) (??:0)
[7] Object::callp(StringName const&, Variant const**, int, Callable::CallError&) (??:0)
[8] Variant::callp(StringName const&, Variant const**, int, Variant&, Callable::CallError&) (??:0)
[9] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) (??:0)
[10] GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) (??:0)
[11] bool Node::_gdvirtual__ready_call<false>() (??:0)
[12] Node::_notification(int) (??:0)
[13] Node::_notificationv(int, bool) (??:0)
[14] CanvasItem::_notificationv(int, bool) (??:0)
[15] Node2D::_notificationv(int, bool) (??:0)
[16] Object::notification(int, bool) (??:0)
[17] Node::_propagate_ready() (??:0)
[18] Node::_propagate_ready() (??:0)
[19] Node::_set_tree(SceneTree*) (??:0)
[20] SceneTree::initialize() (??:0)
[21] OS_LinuxBSD::run() (??:0)
[22] /home/******/Programming/godot/bin/godot.linuxbsd.tools.x86_64(main+0x187) [0x560b9761b210] (??:0)
[23] /usr/lib/libc.so.6(+0x232d0) [0x7f071b4022d0] (??:0)
[24] /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7f071b40238a] (??:0)
[25] /home/******/Programming/godot/bin/godot.linuxbsd.tools.x86_64(_start+0x25) [0x560b9761afb5] (??:0)
-- END OF BACKTRACE --
================================================================
```
I will try to look further into it as my first issue.

**Comment by DarkKilauea** [comment]
This issue is likely in this method, during the first line.

``` cpp
Rect2 NavigationLink2D::_edit_get_rect() const {
	real_t radius = NavigationServer2D::get_singleton()->map_get_link_connection_radius(get_world_2d()->get_navigation_map());

	Rect2 rect(get_start_location(), Size2());
	rect.expand_to(get_end_location());
	rect.grow_by(radius);
	return rect;
}
```

I didn't expect this method to be called when the link wasn't part of the scene tree, so `get_world2d()` isn't guarded by a `is_in_tree()` check.   Adding that guard and returning an empty Rect seems like a reasonable fix.

**Comment by Trashtalk217** [comment]
I've created a fix in https://github.com/godotengine/godot/pull/65444.

**Comment by akien-mga** [comment]
Fixed by #65444.
